src/super_gradients/convert_from_recipe.py

```diff
@@ -0,0 +1,206 @@
+"""
+Entry point for converting recipe file to self-contained train.py file.
+
+General use: python -m super_gradients.convert_from_recipe --config-name="DESIRED_RECIPE".
+For recipe's specific instructions and details refer to the recipe's configuration file in the recipes directory.
+"""
+import argparse
+import collections
+import os.path
+from typing import Tuple, Mapping, Dict
+
+import hydra
+import pkg_resources
+from hydra.core.global_hydra import GlobalHydra
+from omegaconf import DictConfig, OmegaConf, ListConfig
+
+from super_gradients import Trainer
+from super_gradients.common import MultiGPUMode
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.common.environment.omegaconf_utils import register_hydra_resolvers
+from super_gradients.common.environment.path_utils import normalize_path
+from super_gradients.training.utils import get_param
+
+logger = get_logger(__name__)
+
+
+def import_black_or_fail_with_instructions():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        raise ImportError(
+            "Black is not installed. Please install it with `pip install black` and try again. "
+            "If you are using a virtual environment, make sure it is activated."
+        )
+
+
+def import_black_or_install():
+    try:
+        import black
+
+        return black
+    except ImportError:
+        import pip
+
+        pip.main(["install", "black==22.10.0"])
+
+        return import_black_or_fail_with_instructions()
```

Comment:
Are you sure we want this ?
One example is someone running locally without any internet access.

I think we can
- By default, use black like you are doing
- But add an optional flag `--skip-formatting`. When set, we ignore black (and therefore don't need to have internet access to run the script)

Reply:
That is up to discussion. True, w/o internet this would fail and we can gracefully handle that.
Leaving code unformatted IMO is not an option - training_hyperparams otherwise looks like a mess (single-liner)
I really wanted to avoid manual code formatting & indentation logic :|

Reply:
Idea: If someone doesnt have internet, and doesnt have it installed, then we dont format but write a comment at the beginning:
```
# WARNING
# This script was not formated, it may be hard to navigate through the code. 
# To format the text, please 
#     - install black `pip install black=...`
#     - black myscript.py
```

Not mandatory, we can always just leave it as it is currently, and only run if black is installed/can be installed